### PR TITLE
add type_hints to analysis_config

### PR DESCRIPTION
**Pull request by eivindjahren**
**Issue**
Needed for #5214 

## Pre review checklist

- [x] Added appropriate release note label
- [x] PR title captures the intent of the changes, and is fitting for release notes.
- [x] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/_c_wrappers/enkf/analysis_config.py [hunk 1]
```diff
@@ -1,28 +1,28 @@
...
-        analysis_set_var=None,
-        analysis_select=None,
+        alpha: float = 3.0,
+        std_cutoff: float = 1e-6,
+        stop_long_running: bool = False,
+        max_runtime: int = 0,
+        min_realization: int = 0,
+        update_log_path: str = "update_log",
+        analysis_iter_config: Optional[AnalysisIterConfig] = None,
+        analysis_copy: Optional[List[Tuple[str, str]]] = None,
```
Comment (yngve-sk) [inline src/ert/_c_wrappers/enkf/analysis_config.py:23]: Screenshot 2023-07-27 at 15 41 45

File: src/ert/_c_wrappers/enkf/analysis_config.py [hunk 2]
```diff
@@ -1,28 +1,28 @@
...
+        analysis_set_var: Optional[List[Tuple[str, str, str]]] = None,
```
Comment (yngve-sk) [inline src/ert/_c_wrappers/enkf/analysis_config.py:24]: Screenshot 2023-07-27 at 15 41 27

File: src/ert/_c_wrappers/enkf/analysis_config.py [hunk 3]
```diff
@@ -1,28 +1,28 @@
...
+        analysis_select: Optional[str] = None,
```
Comment (yngve-sk) [inline src/ert/_c_wrappers/enkf/analysis_config.py:25]: `Optional[ContextString]`

File: src/ert/_c_wrappers/enkf/analysis_config.py [hunk 4]
```diff
@@ -60,17 +60,17 @@ def _copy_modules(self):
...
 
             module = self.get_module(module_name)
             module.set_var(var_name, value)
 
     @classmethod
-    def from_dict(cls, config_dict) -> "AnalysisConfig":
-        num_realization = config_dict.get(ConfigKeys.NUM_REALIZATIONS, 1)
-        min_realization = config_dict.get(ConfigKeys.MIN_REALIZATIONS, 0)
+    def from_dict(cls, config_dict: ConfigDict) -> "AnalysisConfig":
+        num_realization: int = config_dict.get(ConfigKeys.NUM_REALIZATIONS, 1)  # type: ignore
```
Comment (yngve-sk) [inline src/ert/_c_wrappers/enkf/analysis_config.py:72]: Some of these are Context'ed but some tests also initialize from dict without originating from file, so not sure which one is more practical/correct. Context* if file origin is always assumed.

Screenshot 2023-07-27 at 15 48 03

**Review by yngve-sk — CHANGES_REQUESTED** [review]
Looks good overall, please do a double check and consider whether the typings should be using the context'ed values as they do show up in some places when running this

**Comment by eivindjahren** [comment]
> Looks good overall, please do a double check and consider whether the typings should be using the context'ed values as they do show up in some places when running this

I looked a bit closer at it, and I don't think its a good idea to check the TypeDict portion just yet. I think we can more easily handle this by simplifying the ErrorInfo object a bit now that we are just using the new parser. Then you can have a look at whether that solves the issue.

**Review by yngve-sk — APPROVED** [review]
LGTM🚀

**Pull request merged**